evalml/model_family/model_family.py

```diff
@@ -9,6 +9,7 @@ class ModelFamily(Enum):
     LINEAR_MODEL = 'linear_model'
     CATBOOST = 'catboost'
     EXTRA_TREES = 'extra_trees'
+    DECISION_TREE = 'decision_tree'
```

Comment:
Maybe we can move towards generalizing model families - I see that we have `is_tree_estimator` below and I think it could be a good idea to put all tree based models together. Likewise with all gradient boosted machines! We should file an issue if we like that idea.

Reply:
Yup, this thought also crossed my mind while I was adding this and I agree! It's a bit tricky though, since we currently rely on `ModelFamily` to determine when we're dealing with `XGBoost` and `CatBoost` which both have to be handled differently in quite a few places 🤔